db-scripts/src/main/resources/migration.sql

```diff
@@ -380,4 +380,49 @@ ALTER TABLE `mutation` ADD COLUMN `DRIVER_FILTER_ANNOTATION` VARCHAR(80) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER` VARCHAR(50) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER_ANNOTATION` VARCHAR(80) NULL;
 
-UPDATE info SET DB_SCHEMA_VERSION="2.4.0";
+##version: 2.4.1
+-- ========================== new reference genome genes related tables =============================================
+CREATE TABLE `reference_genome` (
+    `REFERENCE_GENOME_ID` int(4) NOT NULL AUTO_INCREMENT,
+    `SPECIES` varchar(64) DEFAULT NULL,
+    `NAME` varchar(64) DEFAULT NULL,
+    `BUILD_NAME` varchar(64) DEFAULT NULL,
+    `GENOME_SIZE` bigint(20) DEFAULT NULL,
+    `URL` varchar(256) DEFAULT NULL,
+    `RELEASE_DATE` datetime DEFAULT NULL,
+    PRIMARY KEY (`REFERENCE_GENOME_ID`),
+    UNIQUE INDEX `BUILD_NAME_UNIQUE` (`BUILD_NAME` ASC)
+);
+
+INSERT INTO `reference_genome` 
+VALUES (1, 'human', 'hg19', 'GRCh37', NULL, 'http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips', '2009-02-01');
+INSERT INTO `reference_genome` 
+VALUES (2, 'human', 'hg38', 'GRCh38', NULL, 'http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips', '2013-12-01');
+INSERT INTO `reference_genome` 
+VALUES (3, 'mouse', 'mm10', 'GRCm38', NULL, 'http://hgdownload.cse.ucsc.edu//goldenPath/mm10/bigZips', '2012-01-01');
+
+CREATE TABLE `reference_genome_gene` (
+    `ENTREZ_GENE_ID` int(11) NOT NULL,
+    `CYTOBAND` varchar(64) NOT NULL,
+    `EXONIC_LENGTH` int(11) DEFAULT NULL,
+    `START` bigint(20) DEFAULT NULL,
+    `END` bigint(20) DEFAULT NULL,
+    `ENSEMBL_GENE_ID` varchar(64) DEFAULT NULL,
+    `CHR` varchar(5) DEFAULT NULL,
+    `REFERENCE_GENOME_ID` int(4) NOT NULL,
+    PRIMARY KEY (`ENTREZ_GENE_ID`,`REFERENCE_GENOME_ID`),
+    FOREIGN KEY (`REFERENCE_GENOME_ID`) REFERENCES `reference_genome` (`REFERENCE_GENOME_ID`) ON DELETE CASCADE,
+    FOREIGN KEY (`ENTREZ_GENE_ID`) REFERENCES `gene` (`ENTREZ_GENE_ID`) ON DELETE CASCADE
+);
+
+INSERT INTO reference_genome_gene (ENTREZ_GENE_ID, CYTOBAND, EXONIC_LENGTH, CHR, REFERENCE_GENOME_ID)
+(SELECT 
+	ENTREZ_GENE_ID, 
+	CYTOBAND, 
+	LENGTH, 
+	SUBSTRING_INDEX(gene.CYTOBAND,IF(LOCATE('p', gene.CYTOBAND), 'p', 'q'), 1), 
```

Comment:
Given the length of the new CHR definition in reference_genome_gene, this insert will fail for some genes in which the computed CHR is greater than 5.  Maybe increasing this field length is the solution or maybe the substring computation should be adjusted.  I'm attaching a list of distinct CHR computations for consideration.

The long one second from last looks valid for the OMS gene.  What do you think?

[OMS](https://www.ncbi.nlm.nih.gov/gene/619538)

![image](https://user-images.githubusercontent.com/366003/32393440-d856bff8-c0af-11e7-8617-23494943209a.png)

Reply:
@khzhu As requested - here is a list of gene records whith problematic cytobands - most of the records are miRNA (with entrez ids < 0) which I think we can disregard, which leaves only 7.

![image](https://user-images.githubusercontent.com/366003/32453936-ae63afd8-c2eb-11e7-92ee-8133ae9fa199.png)